config/decoder.go

```diff
@@ -119,13 +119,23 @@ func (d *decoder) getGlobalProvider() Provider {
 }
 
 // Sets value to a primitive type.
-func (d *decoder) scalar(childKey string, value reflect.Value, def string) error {
+func (d *decoder) scalar(childKey string, value reflect.Value, def string) (err error) {
 	valueType := value.Type()
 	global := d.getGlobalProvider()
 	var val interface{}
 
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
Out of curiosity, what triggers panic? I've tried testcases and the only failing case was with port being 0, not 4324..